lib/msf/ui/console/history_manager.rb

```diff
@@ -0,0 +1,66 @@
+# -*- coding: binary -*-
+
+module Msf
+module Ui
+module Console
+
+class HistoryManager
+ 
+
+  @@contexts = [{"history_file" => Msf::Config.history_file, "name" => :msfconsole}]
+  
+  def self.inspect
+    "#<HistoryManager stack size: #{@@contexts.length}>"
+  end
+
+  def self.context_stack
+    @@contexts
+  end
+
+  def self.push_context( history_file: nil, name: '')
```

Comment:
To use a consistent data type, the `name:` kwarg should probably be `nil`.

```suggestion
  def self.push_context(history_file: nil, name: nil)
```